### ISSUE TITLE
train error

### ISSUE THREAD

**Issue by felicityji** [torralba-lab/im2recipe-Pytorch#16]
@nhynes   hi , I've executed the code python train.py --batch_size 32 with the batch size 32，
but I got an error:

![screenshot from 2019-01-18 13-38-26](https://user-images.githubusercontent.com/28780306/51367885-950f6780-1b27-11e9-913f-358bfce8e9b3.png)

Can you please advise? thank you very much!

**Comment by felicityji** [comment]
@nhynes  i had only single 8GB GPU and GTX 1080.  What do you suggest here?

**Comment by nhynes** [comment]
Hey @felicityji , your GPU is a bit small for the original im2recipe model, but that's not what's causing the segfault (segfaults happen on the CPU, not GPU). One thing you might try is the [newer im2recipe model](https://github.com/torralba-lab/3j2) which trains faster, to essentially the same accuracy, and without any fiddly alternating. It's not as well documented because nobody has tried to build with it yet, but I'd be more than glad to help you out--just let me know.

**Comment by felicityji** [comment]
Hey @nhynes ,thank you very much for you reply. The above error has been solved and i am very happy to  have a newer im2recipe model. I will try the newer model later. Can you give a simple direction for use of the newer model?  Thank you again for your help!

**Comment by nhynes** [comment]
> simple direction for use of the newer mode

for sure! I'll be able to work on that later this week.

**Comment by felicityji** [comment]
@nhynes   Fine. Thanks so much!

**Comment by felicityji** [comment]
@nhynes 
![image](https://user-images.githubusercontent.com/28780306/52846385-6dbab300-3144-11e9-95a7-69be29514739.png)
![image](https://user-images.githubusercontent.com/28780306/52846400-76ab8480-3144-11e9-91ba-2c4ea627f28c.png)

**Comment by nhynes** [comment]
Hi @felicityji I think that [this script](https://github.com/torralba-lab/im2recipe-Pytorch/blob/master/scripts/rank.py) is what you might be looking for. Please let me know if this is not the case.

**Comment by felicityji** [comment]
@nhynes  Yes, it‘s what i’m looking for. thanks so much!

**Comment by felicityji** [comment]
![image](https://user-images.githubusercontent.com/28780306/54655734-fa6adf00-4afd-11e9-8a10-cf58ed582a33.png)
@nhynes  hi, the cos loss  doesn't look normal at all , it still at 0.0000 and the MedR keep to 500. can you give some advise to solve this problem?  I am looking for your reply!

**Comment by nhynes** [comment]
Looks like training is stalled. Are you using the default settings and dataset?

**Comment by felicityji** [comment]
yes ,  I've executed the code python train.py --batch_size 32 with the batch size 32. And other people are experiencing the same problem!

**Comment by nhynes** [comment]
I guess the gradients are too noisy if the batch size is less than 160 🤷‍♂️ . Try lowering the learning rate?
